jetstream/pull.go

```diff
@@ -638,7 +632,7 @@ func (s *pullSubscription) Next() (Msg, error) {
 			}
 			if errors.Is(err, errDisconnected) {
 				if hbMonitor != nil {
-					hbMonitor.Reset(2 * s.consumeOpts.Heartbeat)
+					hbMonitor.Stop()
```

Comment:
are we sure we're resuming properly after connection is established?

Reply:
Yes, we're resuming hb timer here: https://github.com/nats-io/nats.go/blob/main/jetstream/pull.go#L635

Resetting on disconnect was a bug, it should've been stopped instead